### PR TITLE
Update filter for unevaluated Get Involved step

### LINKED ISSUE #2412: Update view for unevaluated contributions
This view lets us see people with unevaluated contributions. This should be updated to show people with unevaluated "Get Involved" step.

https://amy.carpentries.org/trainings/trainees/?search=&lesson_contribution=on&is_instructor=&training=&training_request=unknown&order_by=

Created automatically from an Asana task.

### PR DESCRIPTION
**Pull request by elichad**
In the interest of time, I'm raising this not against `feature/instructor-checkout-changes` but instead against the branch used in #2431 for review. Will not merge until #2431 is merged.

Fixes #2412.

This PR will:

- update the filter for viewing unevaluated Get Involved submissions (formerly Lesson Contributions)
- add tests for all the trainee filters, because these were missing
- correctly display the date from a Get Involved submission, if present (falling back on the datetime the submission/TrainingProgress was created otherwise)

Screenshot of trainees filtered by unevaluated Get Involved step
Screenshot of hover text "Not evaluated yet Get Involved GitHub Contribution on Thursday 01 June 2023"

### CODE REVIEW

**Comment by froggleston** [comment]
This all looks good to me in terms of code (based on my limited knowledge!) - I'm just wondering if "Not evaluated yet" could be more specific? Is this terminology used by the ITT and would make sense to an AMY user as to what "evaluation" means?

**Comment by maneesha** [comment]
Yes, it will make sense to the ITT.  It means that the trainee has submitted their checkout step but we have not reviewed it internally.  This is the same as how things currently are.

File: amy/trainings/tests/test_filters.py
Comment (pbanaszkiewicz) [inline amy/trainings/tests/test_filters.py:1]: Nice to see this filter getting proper tests!

File: amy/trainings/tests/test_filters.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        )
+
+        # get filterset
+        self.filterset = TraineeFilter({})
+
+        # use a queryset where trainee annotations are available (e.g. is_instructor)
+        # this still includes all Persons
+        self.qs = all_trainees_queryset()
+
+    def tearDown(self) -> None:
```
Comment (pbanaszkiewicz) [inline amy/trainings/tests/test_filters.py:119]: I don't think you have to remove anything from the database. The tests are supposed to be run in nested transaction (pseudo-code):
```
TRANSACTION;
setUp()
TRANSACTION;
runTest1();
ROLLBACK;
TRANSACTION;
runTest2();
ROLLBACK;
ROLLBACK;
```

File: amy/workshops/templatetags/training_progress.py [hunk 2]
```diff
@@ -36,11 +36,17 @@ def progress_description(progress):
     else:
         involvement = ""
 
+    day = (
+        progress.date.strftime("%A %d %B %Y")
+        if progress.date
+        else progress.created_at.strftime("%A %d %B %Y at %H:%M")
+    )
+
     text = "{state} {type}{involvement}<br />on {day}.{notes}".format(
```
Comment (pbanaszkiewicz) [inline amy/workshops/templatetags/training_progress.py:45]: I found this not very readable in your screenshot. Does the type have to be repeated (it's also displayed on the label)?

Comment (pbanaszkiewicz) [reply amy/workshops/templatetags/training_progress.py:45]: Would you like to add tests for this template tag?

**Review by pbanaszkiewicz — APPROVED** [review]
In general it LGTM. I left some minor comments.

File: amy/workshops/templatetags/training_progress.py (hunk 2 above)
Comment (elichad) [reply amy/workshops/templatetags/training_progress.py:45]: Thanks for the readability comment, I agree the wording becomes quite weird in this case - I will ask for some feedback from the ITT on this and see what would work best for them.

I completely overlooked testing of this tag, but I will add some tests either here or in a follow-up PR depending on timing.

**Pull request merged**